File: time/src/main/scala/memeid/time/Posix.scala
package memeid.time

import java.util.concurrent.TimeUnit.MILLISECONDS

/* A posix timestamp. */
trait Posix {

  def value: Long

}

object Posix {

  implicit def apply: Posix = new Posix {
    def value: Long = MILLISECONDS toSeconds System.currentTimeMillis
  }

}

File: modules/memeid-kotlin/src/main/kotlin/memeid/kotlin/time/Posix.kt
package memeid.kotlin.time

import java.util.concurrent.TimeUnit

interface Posix {
  val value: Long
}

val posix: Posix by lazy {
  object : Posix {
    override val value: Long
      get() = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis())
  }
}